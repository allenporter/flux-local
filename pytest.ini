[pytest]
log_level = DEBUG
asyncio_mode = auto
