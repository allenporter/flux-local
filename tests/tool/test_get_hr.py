"""Tests for the flux-local `get hr` command."""

import pytest

from syrupy.assertion import SnapshotAssertion

from . import run_command


@pytest.mark.parametrize(
    ("args"),
    [
        (["-A", "--path", "tests/testdata/cluster"]),
        (["-A", "--path", "tests/testdata/cluster2"]),
        (["-A", "--path", "tests/testdata/cluster3"]),
        (
            [
                "-A",
                "--path",
                "tests/testdata/cluster3",
                "--sources",
                "cluster=tests/testdata/cluster3",
            ]
        ),
        (["-A", "--path", "tests/testdata/cluster4"]),
        (["-A", "--path", "tests/testdata/cluster5"]),
        (["-A", "--path", "tests/testdata/cluster6"]),
        (["-A", "--path", "tests/testdata/cluster7"]),
        (["metallb", "-A", "--path", "tests/testdata/cluster"]),
        (["-n", "metallb", "--path", "tests/testdata/cluster"]),
    ],
    ids=[
        "cluster",
        "cluster2",
        "cluster3-no-source",
        "cluster3",
        "cluster4",
        "cluster5",
        "cluster6",
        "cluster7",
        "all_namespace",
        "name",
    ],
)
async def test_get_hr(args: list[str], snapshot: SnapshotAssertion) -> None:
    """Test test get hr commands."""
    result = await run_command(["get", "hr"] + args)
    assert result == snapshot
