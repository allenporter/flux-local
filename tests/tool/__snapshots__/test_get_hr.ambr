# serializer version: 1
# name: test_get_hr[all_namespace]
  '''
  NAMESPACE    NAME       REVISION    CHART              SOURCE     
  metallb      metallb    4.1.14      metallb-metallb    bitnami    
  
  '''
# ---
# name: test_get_hr[cluster2]
  '''
  NAMESPACE     NAME                    REVISION    CHART                              SOURCE                  
  networking    ingress-nginx           4.5.2       networking-ingress-nginx           ingress-nginx           
  monitoring    kubernetes-dashboard    6.0.0       monitoring-kubernetes-dashboard    kubernetes-dashboard    
  
  '''
# ---
# name: test_get_hr[cluster3-no-source]
  '''
  no HelmRelease objects found in cluster
  
  '''
# ---
# name: test_get_hr[cluster3]
  '''
  NAMESPACE    NAME       REVISION    CHART              SOURCE     
  podinfo      podinfo    None        podinfo-podinfo    podinfo    
  
  '''
# ---
# name: test_get_hr[cluster4]
  '''
  NAMESPACE     NAME                    REVISION    CHART                              SOURCE                  
  monitoring    kubernetes-dashboard    6.0.0       monitoring-kubernetes-dashboard    kubernetes-dashboard    
  
  '''
# ---
# name: test_get_hr[cluster5]
  '''
  NAMESPACE      NAME            REVISION    CHART                       SOURCE          
  metallb        metallb         4.1.14      metallb-metallb             bitnami         
  flux-system    weave-gitops    4.0.22      flux-system-weave-gitops    weave-charts    
  
  '''
# ---
# name: test_get_hr[cluster6]
  '''
  NAMESPACE    NAME        REVISION    CHART               SOURCE      
  default      renovate    37.64.3     default-renovate    renovate    
  
  '''
# ---
# name: test_get_hr[cluster7]
  '''
  NAMESPACE    NAME          REVISION    CHART                  SOURCE            
  database     postgresql    12.7.1      database-postgresql    bitnami-charts    
  
  '''
# ---
# name: test_get_hr[cluster]
  '''
  NAMESPACE      NAME            REVISION    CHART                       SOURCE          
  podinfo        podinfo         6.3.2       podinfo-podinfo             podinfo         
  metallb        metallb         4.1.14      metallb-metallb             bitnami         
  flux-system    weave-gitops    4.0.22      flux-system-weave-gitops    weave-charts    
  
  '''
# ---
# name: test_get_hr[name]
  '''
  NAME       REVISION    CHART              SOURCE     
  metallb    4.1.14      metallb-metallb    bitnami    
  
  '''
# ---
